lib/src/test/kotlin/starknet/provider/ProviderTest.kt

```diff
@@ -814,4 +970,148 @@ class ProviderTest {
         assertNull(receipt.actualFee)
         assertNull(receipt.failureReason)
     }
+
+    @Test
+    fun `get nonce with block number`() {
+        val provider = rpcProvider()
+        val request = provider.getNonce(contractAddress, latestBlock.number)
+        val response = request.send()
+
+        assertNotNull(response)
+    }
+
+    @Test
+    fun `get nonce with block hash`() {
+        val provider = rpcProvider()
+        val request = provider.getNonce(contractAddress, latestBlock.hash)
+        val response = request.send()
+
+        assertNotNull(response)
+    }
+
+    @Test
+    fun `get block with transactions with block tag`() {
+        // FIXME: We should also test for 'pending' tag, but atm they are not supported in devnet
+        val provider = rpcProvider()
+        val request = provider.getBlockWithTxs(BlockTag.LATEST)
+        val response = request.send()
+
+        assertNotNull(response)
+        assertTrue(response is BlockWithTransactionsResponse)
+    }
+
+    @Test
+    fun `get block with transactions with block hash`() {
+        val provider = rpcProvider()
+        val request = provider.getBlockWithTxs(latestBlock.hash)
+        val response = request.send()
+
+        assertNotNull(response)
+        assertTrue(response is BlockWithTransactionsResponse)
+    }
+
+    @Test
+    fun `get block with transactions with block number`() {
+        val provider = rpcProvider()
+        val request = provider.getBlockWithTxs(latestBlock.number)
+        val response = request.send()
+
+        assertNotNull(response)
+        assertTrue(response is BlockWithTransactionsResponse)
+    }
+
+    @Test
+    fun `get state of block with tag`() {
+        val provider = rpcProvider()
+        val request = provider.getStateUpdate(BlockTag.LATEST)
+        val response = request.send()
+
+        assertNotNull(response)
```

Comment:
Shouldn't some more specific tests be used here?

Reply:
What exactly do you have in mind? to check/compare returned fields?

Reply:
If there's a possibility to check fields being "valid" simply, we should do it. If not, let's keep assertNotNull